### PR TITLE
Ported torchvision detection tutorial into sphinx gallery format

### PR DESCRIPTION
**Pull request by vfdev-5**
## Description
- Ported torchvision detection tutorial into sphinx gallery format

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: intermediate_source/torchvision_tutorial.py
Comment (NicolasHug) [reply intermediate_source/torchvision_tutorial.py:109]: Probably this? https://pytorch.org/vision/main/auto_examples/transforms/plot_transforms_getting_started.html#what-are-tvtensors

**Comment by vfdev-5** [comment]
@NicolasHug I can temporarily install torchvision nightly for this tutorial and remove it in the end. Also, people are doing this : https://github.com/pytorch/tutorials/blob/569cf6550176c3ca80fda89fe1bb5ecf38c35cf8/requirements.txt#L34

what would you suggest?

**Comment by NicolasHug** [comment]
I would avoid that. I'm not sure the maintainers of this repo woudl be too happy with potential breakage coming from torchvision nightly.

The situation isn't really comparable to `torchmultimodal`: for torchMM, there is no stable package yet, the only thing that is available is the "nightly" package.


Let's just update the `ipynb` notebook please.

File: intermediate_source/torchvision_tutorial.py
Comment (NicolasHug) [inline intermediate_source/torchvision_tutorial.py]: Just FYI, the search and replace seems to have cause false positives like here.

(Also, ending rst links with 2 backticks `__` instead of 1 may help avoiding issues when links are duplicated... not sure it will be a problem here, but something to consider)

Comment (vfdev-5) [reply intermediate_source/torchvision_tutorial.py]: Just fixed it

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @vfdev-5 , the tutorial and rendered docs all look good as far as I'm concerned, with some easily-addressed points:

- some images do not render properly (like the one below, but there are others)
image

- we write `torchvision.TVTensor` in some places and that doesn't really exists. Maybe we should be using `~` everywhere in order for the links to look shorter, e.g. ```:class:`~torchvision.tv_tensors.TVTensor` ``` or ```:class:`~torchvision.tv_tensors.BoundingBoxes` ```


I'll let @svekars chime regarding the changes to the CI etc.

**Comment by vfdev-5** [comment]
https://docs-preview.pytorch.org/pytorch/tutorials/2540/intermediate/torchvision_tutorial.html#wrapping-up

> Total running time of the script: ( 2 minutes 29.094 seconds)

Addition CI time to build this tutorial

**Comment by vfdev-5** [comment]
@svekars can you please review this PR regarding CI changes, thanks!

**Review by svekars — APPROVED** [review]
Looks great! Thanks for doing this.

**Pull request merged**
